common.go

```diff
@@ -24,6 +24,10 @@ func invMsg(v string) (s string) {
 }
 
 func length(v interface{}) (int, bool) {
+	if v == nil {
+		return 0, true
+	}
```

Comment:
The only time that this would happen would be when someone passes in either a literal `nil` or a variable of type `interface{}` that has both a `nil` value **and** a `nil` type.  It probably makes more sense to explicitly pass `0, false` in this case, because it's not really a valid type for checking length.